lib/spack/spack/version.py

```diff
@@ -59,7 +59,7 @@
                           "(?:[+][0-9A-Za-z-]+)?)")
 
 # Infinity-like versions. The order in the list implies the comparison rules
-infinity_versions = ['develop', 'main', 'master', 'head', 'trunk']
+infinity_versions = ['develop', 'main', 'master', 'head', 'trunk', 'release']
```

Comment:
I'm not familiar with a `release` branch typically being used in this fashion, so I'm concerned about cluttering this definition. It does seem to me we should be able to mark versions as being "latest" without having to update this particular data structure, so I could look into that, but in the meantime would undoing this particular change cause any trouble for you?

The packages you added `@release` to already define `@main`, so given the ordering defined here, it will be highly unlikely that `@release` would ever be chosen unless it was specifically requested.

Reply:
> The packages you added @release to already define @main, so given the ordering defined here, it will be highly unlikely that @release would ever be chosen unless it was specifically requested.

Yes - `spack install petsc@release` is the expected usage. I couldn't get that working without adding  `release` to `infinity_versions`.


Here is the model for release/main we use:

When petsc-3.16.0 is created - its created off `main`. And `release` is synced to it.
Now patches/fixes get added to `release` branch - and petsc-3.16.1, 3.16.2 etc are spun from it.

Until petsc-3.17.0 is spun from 'main' and release synced to main. [here petsc-3.17.1, 3.17.2 etc are spun from release]

So in the spack context - when latest release is 3.16.x - `@release` would correspond to 3.16.9999 [wrt version checks]. I'm not sure how to do that.

But at the minimum - would like to be able to do: `spack install petsc@release` [to install latest contents from release branch]

Reply:
> Yes - spack install petsc@release is the expected usage. I couldn't get that working without adding release to infinity_versions.

Could you print the error you got? When I check this PR out and remove the definition of `release` from `infinity_versions`, I can do `spack stage petsc@release` without an error (and it does in fact appear to check out the release branch). Likewise this appears to work for `py-petsc4py`.

> But at the minimum - would like to be able to do: spack install petsc@release [to install latest contents from release branch]

Since the `petsc@release` version tracks the `release` branch of `petsc`, then this is what you'll get (Spack will pull whatever is on the tip of the release branch and install that).

Reply:
> Could you print the error you got?
```
[balay@pj01 spack]$ spack install petsc@release
==> Error: petsc@release is unsatisfiable, conflicts are:
  External package version does not satisfy external spec
  no version satisfies the given constraints
```